components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/main/java/org/wso2/carbon/apimgt/rest/api/endpoint/registry/impl/RegistriesApiServiceImpl.java

```diff
@@ -161,9 +176,35 @@ public Response createRegistryEntry(String registryId, RegistryEntryDTO registry
                         " is not found", log);
             }
             ResourceFile definitionFile;
+            if ((definitionFileInputStream != null || registryEntry.getDefinitionUrl() != null) &&
+                    registryEntry.getDefinitionType() == null) {
+                RestApiUtil.handleBadRequest("Missing definitionType of the new registry " +
```

Comment:
the message seems to be too long and a bit confusing. "Missing definitionType parameter" would be sufficient.

Reply:
updated